projects/asn1-proto/asn1_proto_converter.cc

```diff
@@ -0,0 +1,166 @@
+#include "asn1_proto_converter.h"
+
+namespace asn1_proto {
+
+/* Returns number of bytes needed to encode num
+ * Where num is a variable-length integer with
+ * no leading zeros
+ */
+uint8_t ASN1ProtoConverter::GetNumBytes(size_t num) {
+  uint8_t shift = sizeof(num);
+  while (shift != 0) {
+    if (((num >> ((shift - 1) * 8)) & 0xFF) != 0) {
+      return shift;
+    }
+    shift--;
+  }
```

Comment:
```suggestion
  for (uint8_t shift = sizeof(num); shift != 0; --shift) {
    if (((num >> ((shift - 1) * 8)) & 0xFF) != 0) {
      return shift;
    }
  }
```

Could be a simpler way to do this.